flocker/node/agents/cinder.py

```diff
@@ -0,0 +1,237 @@
+        # again. One day we hope this won't be necessary.
+        # See Rackspace support ticket: 150422-ord-0000495'
+        self.volume_manager.set_metadata(created_volume, metadata)
+        # Use requested volume here, because it has the desired metadata.
+        return _blockdevicevolume_from_cinder_volume(requested_volume)
+
+    def list_volumes(self):
+        """
+        Return ``BlockDeviceVolume`` instances for all the Cinder Volumes that
+        have the expected ``cluster_id`` in their metadata.
+
+        See: http://docs.rackspace.com/cbs/api/v1.0/cbs-devguide/content/GET_getVolumesDetail_v1__tenant_id__volumes_detail_volumes.html # noqa
+        """
+        volumes = []
+        for cinder_volume in self.volume_manager.list():
+            if _is_cluster_volume(self.cluster_id, cinder_volume):
+                volumes.append(
+                    _blockdevicevolume_from_cinder_volume(cinder_volume)
+                )
+        return volumes
+
+    def resize_volume(self, blockdevice_id, size):
+        pass
+
+    def attach_volume(self, blockdevice_id, host):
+        pass
+
+    def detach_volume(self, blockdevice_id):
+        pass
+
+    def destroy_volume(self, blockdevice_id):
+        pass
+
+    def get_device_path(self, blockdevice_id):
+        pass
+
+
+def _is_cluster_volume(cluster_id, cinder_volume):
+    """
+    :param UUID cluster_id: The uuid4 of a Flocker cluster.
+    :param Volume cinder_volume: The Volume with metadata to examine.
+    :return: ``True`` if ``cinder_volume`` metadata has a
+        ``CLUSTER_ID_LABEL`` value matching ``cluster_id`` else ``False``.
+    """
+    actual_cluster_id = cinder_volume.metadata.get(CLUSTER_ID_LABEL)
+    if actual_cluster_id is not None:
+        actual_cluster_id = UUID(actual_cluster_id)
+        if actual_cluster_id == cluster_id:
+            return True
+    return False
+
+
+def _blockdevicevolume_from_cinder_volume(cinder_volume):
+    """
+    :param Volume cinder_volume: The ``cinderclient.v1.volumes.Volume`` to
+        convert.
+    :returns: A ``BlockDeviceVolume`` based on values found in the supplied
+        cinder Volume.
+    """
+    return BlockDeviceVolume(
+        blockdevice_id=unicode(cinder_volume.id),
+        size=int(GB(cinder_volume.size).to_Byte().value),
+        host=None,
+        dataset_id=UUID(cinder_volume.metadata[DATASET_ID_LABEL])
+    )
+
+
+def rackspace_cinder_client(**kwargs):
+    """
+    Create a Cinder API client capable of authenticating with Rackspace and
+    communicating with their Cinder API.
+
+    :param unicode username: A RackSpace API username.
+    :param unicode api_key: A RackSpace API key.
+    :param unicode region: A RackSpace region slug.
+    :return: A ``cinderclient.v1.clien.Client`` instance with a ``volumes``
+        attribute that conforms to ``ICinderVolumeManager``.
+    """
+    username = kwargs.pop('username')
```

Comment:
Because the `acceptance.yml` contains a `keyname` variable, which isn't useful here, but I wanted to just pass the whole dictionary in:
 * http://doc-dev.clusterhq.com/gettinginvolved/acceptance-testing.html#rackspace